bin/init.ml

```diff
@@ -90,15 +79,40 @@ let context_cwd : Init_context.t Term.t =
   Scheduler.go ~common:common_term ~config (fun () ->
       Memo.run (Init_context.make path))
 
-let public : Component.Options.public_name option Term.t =
+module Public_name = struct
+  type t =
+    | Use_name
+    | Public_name of Dune_lang.Atom.t
```

Comment:
How about using `Lib_name.t` for this?

Reply:
Yes, I've got something in the works to use the proper types instead of atoms (this type is only moved in this PR)